spec/support/e2e_setup.rb

```diff
@@ -26,10 +29,20 @@ def set_hidden_field_value(id, value)
     find_field(id, type: :hidden).set(value)
   end
 
+  def mui_choose(choice, from:)
+    label = find('label', text: from)
+    id = label['id']
+    within("[aria-labelledby='#{id}']") do
+      choose(choice)
+    end
+  end
+
   def mui_select(choice, from:)
     label = find('label', text: from)
     id = label['for']
-    find("##{id}").click
+    # we seem to have invalid ids such as "3.917A.1"
+    # find("##{id}").click
+    find("[id='#{id}']").click
```

Comment:
Oh I see, I didn't realize they were invalid. IDs must start with A-Za-z, that's the only issue right? The ID gets set to the unique `link_id` in the form definition. Maybe we should do something like `LINK_ID_${linkId}` instead. (Not requesting a change, just pointing out)

Reply:
That's right, I think they just need to start with a letter (you can also escape them too). To clarify, the ids are valid in HTML, they just can't be used in CSS selectors